### ISSUE TITLE
Mitre buffer returns invalid geometry

### ISSUE THREAD

**Issue by mwtoews** [libgeos/geos#856]
This appears to be a minor robustness corner case with a miter buffer using the following shapely script:
```python
from shapely import geos, wkt

print("geos", geos.geos_version_string)

A_wkt = """\
    POLYGON ((-23.989123360549296 73.1287474328027,
              -22.537997105552297 94.06368412079055,
              -18.796973600895146 93.80437130274495,
              -17.80121237894408 108.16990157009043,
              -21.542235883601226 108.42921438813606,
              -20.967403753721864 116.7221345967023,
              -4.728530705460814 116.7221568196225,
              -7.82790182044367 72.00851605865441,
              -23.989123360549296 73.1287474328027))"""

# this is the only invalid buffer example
A = wkt.loads(A_wkt)
B = A.buffer(10, join_style=2)
print("main bug:", B.is_valid)

# the remaining examples are valid
A = wkt.loads(A_wkt)
B = A.buffer(10 + 1e-14, join_style=2)
print("small buffer distance adjustment:", B.is_valid)

Ap = wkt.loads(wkt.dumps(A, rounding_precision=14))
B = Ap.buffer(10, join_style=2)
print("rounding_precision=14:", B.is_valid)
```
Note that the error only happens in the machine epsilon (1e-14) scale of floats, suggesting a robustness issue. While I haven't tested all GEOS versions, I see the same behaviour with 3.9.1, so it isn't new. I don't see the behaviour with JTS, but here is a JTS figure showing valid input A in blue and invalid B in red.

![image](https://user-images.githubusercontent.com/895458/228672821-435d65cd-ae75-49bc-849e-077c90823b54.png)

Originally reported in https://github.com/shapely/shapely/issues/1792

**Comment by pramsey** [comment]
In general we aren't guaranteeing valid output from Buffer, though that might change if/when it gets moved off the old graph structure and onto the NG foundation. However, no great hurry on that. As mentioned in the other similar ticket, we could post-check validity and make valid, but... actually in this case you'd end up with a spurious hole, so the valid output wouldn't actually be any better.

**Comment by pramsey** [comment]
Still seeing this? I just tried it directly on main, and it passed.
```c++
// Buffer produces invalid output
// https://github.com/libgeos/geos/issues/856
template<>
template<>
void object::test<23>()
{
    auto geom = wktreader.read("POLYGON ((-23.989123360549296 73.1287474328027, -22.537997105552297 94.06368412079055,-18.796973600895146 93.80437130274495,-17.80121237894408 108.16990157009043,-21.542235883601226 108.42921438813606,-20.967403753721864 116.7221345967023,-4.728530705460814 116.7221568196225,-7.82790182044367 72.00851605865441,-23.989123360549296 73.1287474328027))");

    BufferParameters bp;
    bp.setJoinStyle(BufferParameters::JOIN_MITRE);
    BufferOp op(geom.get(), bp);

    auto result = op.getResultGeometry(10);
    ensure(result->isValid());
}
```

**Comment by mwtoews** [comment]
Running a bunch of bisections, I can see that this was resolved with bf18ba807b775f429bbbc62d5eed4fe24390cee7 (#937).
Possible actions are to add this test (to `tests/unit/operation/buffer/BufferParametersTest.cpp`), backport, and/or simply close this issue as resolved only for next release.

**Comment by mwtoews** [comment]
Closed via backports and the above test (only on `main` with 0a8fc37a965bd29d61d14249ba22a2ae66fd58fa).

The backports are made back to `3.9`, when DoubleDouble math ops were added (bed36f15c780057ae9b83eb9cd2e8ef6a9ada498).
Here are the backport branch commit details:

- [`3.12`](https://github.com/libgeos/geos/commits/3.12) via 037ccb9560f06c05c660666dcb6eac97caf908dd, 540a6aad1e909284385b6403d7a61d56c4deeab7
- [`3.11`](https://github.com/libgeos/geos/commits/3.11) via 1d75f8f8f34c74bab1098798b9b08d4d097c5aa6, 04e7cbed5e3f49e4338f5aa680296c34079e192c
- [`3.10`](https://github.com/libgeos/geos/commits/3.10) via 6d0c831dfbbb861cf3445257a322bbb21600471e, 078d549132cf9044f0aaf182480c3e7b3ce01a8e
- [`3.9`](https://github.com/libgeos/geos/commits/3.9) via c7769ea3b1fb383eab631f21ef872ffffbe1eec8, d1d221f0d412db83f606125f4625faaafcd6979e
